rust/arrow/src/array/equal/list.rs

```diff
@@ -151,8 +151,8 @@ pub(super) fn list_equal<T: OffsetSizeTrait>(
             let lhs_pos = lhs_start + i;
             let rhs_pos = rhs_start + i;
 
-            let lhs_is_null = !get_bit(lhs_null_bytes, lhs_pos);
-            let rhs_is_null = !get_bit(rhs_null_bytes, rhs_pos);
+            let lhs_is_null = !get_bit(lhs_null_bytes, lhs_pos + lhs.offset());
```

Comment:
There is another lhs/rhs mixup a few lines above (148), github won't let me comment on that line:
```
        // get a ref of the parent null buffer bytes, to use in testing for nullness
        let lhs_null_bytes = rhs_nulls.unwrap().as_slice();
                             ^
```

Reply:
Confirmed, at line 147

Reply:
Great catch. Pushed a fix to it.